x/delayedack/keeper/invariants.go

```diff
@@ -1,64 +1,71 @@
+}
 
-// RegisterInvariants registers the delayedack module invariants
-func (k Keeper) RegisterInvariants(ir sdk.InvariantRegistry) {
-	// INVARIANTS DISABLED SINCE LAZY FINALIZATION FEATURE
+func RegisterInvariants(ir sdk.InvariantRegistry, k Keeper) {
+	invs.RegisterInvariants(types.ModuleName, ir, k)
 }
 
-// PacketsFinalizationCorrespondsToFinalizationHeight checks that all rollapp packets stored are set to
-// finalized status for all heights up to the latest height.
-func PacketsFinalizationCorrespondsToFinalizationHeight(k Keeper) sdk.Invariant {
-	return func(ctx sdk.Context) (string, bool) {
-		var (
-			broken bool
-			msg    string
-		)
+// DO NOT DELETE
+func AllInvariants(k Keeper) sdk.Invariant {
+	return invs.All(types.ModuleName, k)
+}
 
-		for _, rollapp := range k.rollappKeeper.GetAllRollapps(ctx) {
-			msg = k.checkRollapp(ctx, rollapp)
-			if msg != "" {
-				msg += fmt.Sprintf("rollapp: %s, msg: %s\n", rollapp.RollappId, msg)
-				broken = true
-			}
+func InvariantProofHeight(k Keeper) uinv.Func {
+	return uinv.AnyErrorIsBreaking(func(ctx sdk.Context) error {
+		var errs []error
+		for _, ra := range k.rollappKeeper.GetAllRollapps(ctx) {
+			err := k.checkRollapp(ctx, ra)
+			err = errorsmod.Wrapf(err, "rollapp: %s", ra.RollappId)
+			errs = append(errs, err)
 		}
-
-		return sdk.FormatInvariant(types.ModuleName, routeFinalizedPacket, msg), broken
-	}
+		return errors.Join(errs...)
+	})
 }
 
-func (k Keeper) checkRollapp(ctx sdk.Context, rollapp rtypes.Rollapp) (msg string) {
-	// will stay 0 if no state is found
+func (k Keeper) checkRollapp(ctx sdk.Context, ra rtypes.Rollapp) error {
+	// will stay 0 if no state is ok
 	// but will still check packets
 	var latestFinalizedHeight uint64
 
-	latestFinalizedStateIndex, found := k.rollappKeeper.GetLatestFinalizedStateIndex(ctx, rollapp.RollappId)
-	if !found {
-		return
+	latestFinalizedStateIndex, ok := k.rollappKeeper.GetLatestFinalizedStateIndex(ctx, ra.RollappId)
+	if !ok {
+		return nil
 	}
 
-	latestFinalizedStateInfo := k.rollappKeeper.MustGetStateInfo(ctx, rollapp.RollappId, latestFinalizedStateIndex.Index)
+	latestFinalizedStateInfo := k.rollappKeeper.MustGetStateInfo(ctx, ra.RollappId, latestFinalizedStateIndex.Index)
 	latestFinalizedHeight = latestFinalizedStateInfo.GetLatestHeight()
 
-	packets := k.ListRollappPackets(ctx, types.ByRollappID(rollapp.RollappId))
-	for _, packet := range packets {
-		if packet.ProofHeight > latestFinalizedHeight && packet.Status == commontypes.Status_FINALIZED {
-			return fmt.Sprintf("rollapp packet for the height should not be in finalized status. height=%d, rollapp=%s, status=%s\n",
-				packet.ProofHeight, packet.RollappId, packet.Status)
-		}
+	packets := k.ListRollappPackets(ctx, types.ByRollappID(ra.RollappId))
+	var errs []error
+	for _, p := range packets {
+		err := k.checkPacket(p, latestFinalizedHeight)
+		err = errorsmod.Wrapf(err, "packet: %s", p.RollappId)
+		errs = append(errs, err)
+	}
+	return errors.Join(errs...)
```

Comment:
```suggestion
	var errs error
	for _, p := range packets {
		err := k.checkPacket(p, latestFinalizedHeight)
		err = errors.Join(errorsmod.Wrapf(err, "packet: %s", p.RollappId))
	}
	return errs
```

so you don't append `nil`s for no reason

Reply:
that's not how you are supposed to use it
nil is fine